index.bs

```diff
@@ -502,9 +504,11 @@ The <dfn method for=Window>getScreenDetails()</dfn> method steps are:
 
 1. Run the following steps [=/in parallel=]:
 
+    1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=policy-controlled feature=] named "[=permission-policy/window-management=]", then [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
+
     1. Let |permissionState| be [=/request permission to use=] `"window-management"`.
 
-    1. If |permissionState| is "{{PermissionState/denied}}" then [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
+    1. If |permissionState| is "{{PermissionState/denied}}", then [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
```

Comment:
Before rejecting a promise (or doing any of the other steps below) you need to "[queue a global task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-global-task) on the [relevant global object](https://html.spec.whatwg.org/multipage/webappapis.html#concept-relevant-global) of [this](https://webidl.spec.whatwg.org/#this) using the [window placement task source](https://w3c.github.io/window-placement/#window-placement-task-source)" so that you are back on the main thread. Otherwise you are modifying things like `this.screenDetails` in a way that could race with executing JavaScript.

Reply:
Done, thanks for catching that! Hopefully it's more correct now, even if it could be refined.